### ISSUE TITLE
Add Aerospike as a state store

### ISSUE THREAD

**Issue by abhirockzz** [dapr/components-contrib#195]
## Describe the proposal

Add support for [Aerospike](https://www.aerospike.com/) - a NoSQL data store

**Comment by abhirockzz** [comment]
Sample configuration
```
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: aerospike
spec:
  type: state.aerospike
  metadata:
    - name: hosts
      value: "localhost:3000"
    - name: namespace
      value: "test"
    - name: set
      value: "set1"
```

**Comment by abhirockzz** [comment]
Sample operations:

Adding records (assuming `dapr` is running on port `3500` state store name in config is `aerospike` )

```
curl -X POST http://localhost:3500/v1.0/state/aerospike \
  -H "Content-Type: application/json" \
  -d '[
        {
          "key": "person1",
          "value": {
            "email": "person1@gmail.com",
            "name":"john"
          }
        }
      ]'

curl -X POST http://localhost:3500/v1.0/state/aerospike \
  -H "Content-Type: application/json" \
  -d '[
        {
          "key": "person2",
          "value": {
            "email": "person2@yahoo.com",
            "name":"jack",
            "phone": {
              "work": "123456789",
              "home": "987654321"
            }
          }
        }
      ]'

curl -X POST http://localhost:3500/v1.0/state/aerospike \
  -H "Content-Type: application/json" \
  -d '[
        {
          "key": "person3",
          "value": {
            "email": "person3@hotmail.com",
            "name":"brad",
            "skills": ["go","java","rust"]
          }
        }
      ]'
```

Getting record (assuming `dapr` is running on port `3500` state store name in config is `aerospike` )

```
curl http://localhost:3500/v1.0/state/aerospike/person1
curl http://localhost:3500/v1.0/state/aerospike/person2
curl http://localhost:3500/v1.0/state/aerospike/person3
curl  http://localhost:3500/v1.0/state/aerospike/blabla
```

Update record with ETag

```
curl -i http://localhost:3500/v1.0/state/aerospike/person1 | grep Etag


curl -X POST http://localhost:3500/v1.0/state/aerospike \
  -H "Content-Type: application/json" \
  -d '[
        {
          "key": "person1",
          "value": {
            "email": "person1@gmail.com",
            "name":"johndoe"
          },
          "etag":"1"
        }
      ]'

curl http://localhost:3500/v1.0/state/aerospike/person1
```

Wrong etag

```
curl -X POST http://localhost:3500/v1.0/state/aerospike \
  -H "Content-Type: application/json" \
  -d '[
        {
          "key": "person1",
          "value": {
            "email": "person1@gmail.com",
            "name":"johndoe"
          },
          "etag":"4242"
        }
      ]'
```

Delete record

```
curl -X DELETE http://localhost:3500/v1.0/state/aerospike/person1
curl http://localhost:3500/v1.0/state/aerospike/person1
```

Delete record with Etag

```
curl -i http://localhost:3500/v1.0/state/aerospike/person2 | grep Etag
curl -X DELETE http://localhost:3500/v1.0/state/aerospike/person2 -H "If-Match: <REPLACE_ETAG_VALUE>"
curl http://localhost:3500/v1.0/state/aerospike/person2
```

Delete record with wrong Etag

```
curl -X DELETE http://localhost:3500/v1.0/state/aerospike/person3 -H "If-Match: 424242"
```

**Comment by yaron2** [comment]
Closed by #196
